### ISSUE TITLE
Custom Metrics?

### ISSUE THREAD

**Issue by doubianimehdi** [pycaret/pycaret#176]
Hi,

Is there a way to specify our own metric for regression ?

Thanks for your answer !

**Comment by pycaret** [comment]
@doubianimehdi No at this point. Can you give some example and use-cases? that will help us priotize this. Thanks.

**Comment by doubianimehdi** [comment]
Yes, in this competition , TreNds Neuroimaging , this person used PyCaret : https://www.kaggle.com/rohitsingh9990/trends-pycaret-training-inference but the metric used to improve the models was the MAE , this competition used a custom metrics called feature weighted normalized error and the formula of this metric is : 
ubmissions are scored using feature-weighted, normalized absolute errors.

score=∑fwf(∑i|yf,i−y^f,i|∑iy^f,i)
where yf,i is the ith observation of feature f, y^f,i is the corresponding ground truth for that observation, and wf is a weighting given to each feature. The weights are [.3, .175, .175, .175, .175] corresponding to features [age, domain1_var1, domain1_var2, domain2_var1, domain2_var2].

A small percentage of values are missing from the ground truth. These are skipped in the calculation. You should, though, make predictions for every row in the submission file.

https://www.kaggle.com/c/trends-assessment-prediction/overview/evaluation

**Comment by bhishanpdl** [comment]
To add to this, I was interested in using Quadratic Weighted Kappa for multiclass classification.

I assume default kappa is cohens kappa with linear weights, but I am looking for quadratic weights.

The use case if kaggle competition for Prudential Insurance.
- https://www.kaggle.com/c/prudential-life-insurance-assessment/overview/evaluation


If there is already a way to assign quadratic weight please let me know. Thanks.

**Comment by oguzhangur96** [comment]
Hi, thanks for this beautiful module.

In some cases customers have their custom metric to measure human successs while you have to surpass humans by using machine learning. For example it is common, almost traditional for sales persons to predict future sales to/of customers they are responsible to sell in my country.

One might cannot use this custom metric to optimize inner parameters of a model but it might be useful to pass as a metric for hyperparameter optimization.

Autogluon has this feature, they also utilize their default metrics using same function **make_scorer** you could find in below link.

[https://github.com/awslabs/autogluon/blob/master/autogluon/utils/tabular/metrics/__init__.py](url)

I could work & support you about this feature if needed.

**Comment by pycaret** [comment]
@oguzhangur96 @doubianimehdi 

`custom_scorer` parameter is added to `tune_model` in `classification.py` and `regression.py`. You can now pass your own optimizer created using `sklearn.make_scorer` in `tune_model`.

Please sync to `master` branch to use this feature. If you are having issues in the use, please open another issue.
